backend/src/search/SearchIndexBuilder.ts

```diff
@@ -0,0 +1,342 @@
+import MiniSearch from 'minisearch';
+import { Injectable } from '@nestjs/common';
+import { Neo4jService } from 'nest-neo4j/dist';
+import { neo4jReturnEdge, neo4jReturnNode } from '../config/commonFunctions';
+import { Node, Edge } from '../shared/entities';
+import { parseNeo4jEntityInfo } from '../schema/parseNeo4jEntityInfo';
+import { EdgeType, EntityType, NodeType } from '../shared/schema';
+import { SearchIndex } from './SearchIndex';
+import { AsyncLazy } from '../shared/utils';
+import { IndexEntry } from './IndexEntry';
+
+/**
+ * Flattens the specified string array.
+ */
+function flattenArray(array: string[]): string {
+  return `[${array.join(', ')}]`;
+}
+
+/**
+ * Converts the specified value to string or return null if the value cannot be converted.
+ * @param value The value to stringify.
+ * @returns The stringified version of the value, or null of the value cannot be converted.
+ */
+function convertToString(value: unknown): string | null {
+  // If the value is a string, we are done.
+  if (typeof value === 'string') {
+    return value;
+  }
+
+  // If the value is an array, convert each element and combine them via flattenArray.
+  if (Array.isArray(value)) {
+    if (value.length > 1) {
+      return flattenArray(
+        <string[]>(
+          value.map((entry) => convertToString(entry)).filter((entry) => entry)
```

Comment:
This will remove empty strings, but I thinks this is also fine right?

Reply:
Yes, that's intended.